src/main/java/org/opentripplanner/routing/impl/ATLFareServiceImpl.java

```diff
@@ -0,0 +1,391 @@
+            // so we only check if it's valid if the transfer is going to be used.
+            if (!transferClassification.type.equals(TransferType.NO_TRANSFER)) {
+                // Consider the conditions under which this transfer will no longer be valid.
+                if (transferClassification.type.equals(TransferType.END_TRANSFER)) return false;
+                else if (transferUseTime >= transferStartTime + transferWindow * 60L) return false;
+                else if (rides.size() > maxRides) return false;
+            }
+
+            // The transfer is valid for this ride, so create a fare object.
+            // TODO: Change this fare object out for the one on the Ride object when Fare-by-leg is added
+            fares.add(fare);
+
+            if (transferClassification.type.equals(TransferType.NO_TRANSFER)) {
+                fare.addFare(fareType, getMoney(currency, defaultFare));
+                // Full fare is charged, but transfer is still valid.
+                // Ride is not added to rides list since it doesn't count towards transfer limit.
+                // NOTE: Rides and fares list will not always be in sync because of this.
+                return true;
+            }
+
+            // All conditions below this point "use" the transfer, so we add the ride.
+            rides.add(ride);
+            if (transferClassification.type.equals(TransferType.FREE_TRANSFER)) {
+                fare.addFare(fareType, getMoney(currency, 0));
+                lastFareWithTransfer = defaultFare;
+                return true;
+            } else if (transferClassification.type.equals(TransferType.TRANSFER_PAY_DIFFERENCE)) {
+                float newCost = 0;
+                if (defaultFare > lastFareWithTransfer) {
+                    newCost = defaultFare - lastFareWithTransfer;
+                }
+                fare.addFare(fareType, getMoney(currency, newCost));
+                lastFareWithTransfer = defaultFare;
+                return true;
+            } else if (transferClassification.type.equals(TransferType.TRANSFER_WITH_UPCHARGE)) {
+                fare.addFare(fareType, getMoney(currency, (float) transferClassification.upcharge / 100));
+                lastFareWithTransfer = defaultFare;
+                return true;
+            }
+            return true;
+        }
+
+        public float getTotal() {
+            int total = 0;
+            for (Fare f : fares) {
+               total += f.getFare(fareType).getCents();
+            }
+            return (float) total / 100;
+        }
+    }
+
+    /**
+     * Get the ride price for a single leg. If testing, this class is being called directly so the required agency cash
+     * values are not available therefore the default test price is used instead.
+     */
+    private float getRidePrice(Ride ride, Fare.FareType fareType, Collection<FareRuleSet> fareRules) {
+        if (IS_TEST) {
```

Comment:
I don't think the IS_TEST variable is good design. I would instead override the method in the test. I've done this is in the following commit: https://github.com/ibi-group/OpenTripPlanner/commit/34f41587bedbffcf77578f212ad13e58446a39ef

If you like that, you can cherry-pick it into this branch.

Reply:
Very nice!! Thank you for that.